File: Programmers/코딩 기초 트레이닝/정수를 나선형으로 배치하기.py
def solution(n):
    answer = []
    # 초기화
    for i in range(n):
        temp = []
        for j in range(n):
            temp.append(0)
        answer.append(temp)
    
    direction = 0 #0,1,2,3
    row = 0
    col = 0
    x = 0
    
    min_row = 1
    max_row = n-1
    
    min_col = 0
    max_col = n-1
    
    while True:
        if min_row > max_row and min_col > max_col:
            break
        x += 1
        answer[row][col] = x
        # 오른쪽일 경우
        if direction == 0:
            if col == max_col:
                direction = 1
                max_col -= 1
                row += 1
            else:
                col += 1
        # 아래일 경우
        elif direction == 1:
            if row == max_row:
                direction = 2
                max_row -= 1
                col -= 1
            else:
                row += 1
        # 왼쪽일 경우
        elif direction == 2:
            if col == min_col:
                direction = 3
                min_col += 1
                row -= 1
            else:
                col -= 1
        # 위쪽일 경우
        elif direction == 3:
            if row == min_row:
                direction = 0
                min_row += 1
                col += 1
            else:
                row -= 1   
    return answer

File: Programmers/코딩 기초 트레이닝/정수를 나선형으로 배치하기.js
function solution(n) {
    var answer = [];
    for(let i=0; i<n; i ++){
        let temp = [];
        for(let j=0; j<n; j++){
            temp.push(0);
        }
        answer.push(temp);
    }
    
    let x = 0;
    let y = 0;
    
    let max_x = n-1;
    let max_y = n-1;
    let min_x = 0;
    let min_y = 1;
    
    let d = 0; //0: 오른쪽, 1: 아래, 2: 왼쪽, 3: 위
    
    for(let i=1; i<=n*n; i++){
        answer[y][x] = i;
        if(i != n*n){
            //오른쪽 으로 갈때
            if(d == 0){
                if(max_x == x){
                    max_x -= 1;
                    y += 1
                    d = 1;
                }
                else{
                    x += 1
                } 
            }
            else if (d == 1){
                if(max_y == y){
                    max_y -= 1;
                    x -= 1;
                    d = 2
                }
                else{
                    y += 1
                }
            }
            else if (d == 2){
                if(min_x == x){
                    min_x += 1;
                    y -= 1;
                    d = 3
                }
                else{
                    x -= 1
                }
            }
            else if (d == 3){
                if(min_y == y){
                    min_y += 1;
                    x += 1;
                    d = 0
                }
                else{
                    y -= 1
                }
            }
        }
    }
    
    
    return answer;
}